### ISSUE TITLE
When docker run, getting error "Plugin reinitialization required. Please run "terraform init".

### ISSUE THREAD

**Issue by akoranne** [cloudfoundry-incubator/bosh-google-cpi-release#222]
I am trying to install bosh on gcp. I followed the steps as documented. 
When I try to run docker image, I get error. Below is the stacktrace.

```
bosh-standalone:~/workspace/bosh-google-cpi-release/docs/bosh$ docker run -i -t \
>   -e "GOOGLE_CREDENTIALS=${GOOGLE_CREDENTIALS}" \
>   -v `pwd`:/$(basename `pwd`) \
>   -w /$(basename `pwd`) \
>   hashicorp/terraform:light plan \
>     -var service_account_email=${service_account_email} \
>     -var project_id=${project_id} \
>     -var region=${region} \
>     -var zone=${zone} \
>     -var baseip=${base_ip} \
>     -var network_project_id=${xpn_host_project_id-$project_id}
Plugin reinitialization required. Please run "terraform init".
Reason: Could not satisfy plugin requirements.
Plugins are external binaries that Terraform uses to access and manipulate
resources. The configuration provided requires plugins which can't be located,
don't satisfy the version constraints, or are otherwise incompatible.
1 error(s) occurred:
* provider.google: no suitable version installed
  version requirements: "(any version)"
  versions installed: none
Terraform automatically discovers provider requirements from your
configuration, including providers used in child modules. To see the
requirements and constraints from each module, run "terraform providers".
error satisfying plugin requirements
bosh-standalone:~/workspace/bosh-google-cpi-release/docs/bosh$
```

```
bosh-standalone:~/workspace/bosh-google-cpi-release/docs/bosh$ docker images
REPOSITORY            TAG                 IMAGE ID            CREATED             SIZE
hashicorp/terraform   light               ad2835110518        2 days ago          82.14 MB
bosh-standalone:~/workspace/bosh-google-cpi-release/docs/bosh$
```

The docker image is 2 days old. Not sure if I am doing something incorrect, or there is an issue with the image.
Can you please take a look and help me point in the right direction?

Thanks,
--ajay

**Comment by neil-aitken** [comment]
also facing this error

**Comment by akoranne** [comment]
I finally figured it out. The docker image  `terraform` does not have the right plugin. 
You either have to run the docker image standalone, in interactive mode, and download the google plugin. Or get away from the docker image and just install `terraform` binary on my local, target my gcp creds and run the terraform script. That is what I did.

**Comment by johnsonj** [comment]
Thanks @akoranne! I think this is till an issue others will face so we should look to fix the docs.

I think this is related to terraform making the provides modular.

**Comment by DocBradfordSoftware** [comment]
I worked around by using an older image (0.9.9).

**Comment by kichooo** [comment]
So docker terraform is now google cloud - unfriendly

sad :(

Is there some nice and clean solution for it? Installing terraform on all the hosts obviously isn't valid. I will probably have ci build my own custom terraform docker with all the plugins I need preinstalled, but this is an annoyance.

**Comment by johnsonj** [comment]
This should be resolved by #229 where we can use the latest terraform version. please re-open if this is still an issue or pops up again.
